torch/utils/_sympy/functions.py

```diff
@@ -89,6 +89,44 @@
 ]
 
 
+# Return (c,symbol) for an expression of form c*symbol, note that symbol is seen as 1*symbol.
+# Return none if the expression is not symbol or c*symbol.
+def is_mult_const_symbol(expr):
+    if expr.is_symbol:
+        return (1, expr)
+    if (
+        expr.is_Mul
+        and len(expr._args) == 2
+        and expr._args[0].is_integer
+        and expr._args[1].is_symbol
+    ):
+        return (expr._args[0], expr._args[1])
+    return None
+
+
+# A base binary summation is of the form c*a + c*b or a+b,  where a!=b.
```

Comment:
I read the PR description and did not understand, I understand what this function does but this seems like to be a strange canonical form to be designing for. Maybe sym_node.py will make it more clear.

Reply:
mhmm this is just the base case ( the smallest expression that satisfy the property, I can move this to sym_node.py .I think as i generalize this (as i mentioned in the other comment) it would be c1*a + c2*b or a+b.